src/uris/deepLinks/deepLinkService.ts

```diff
@@ -1077,7 +1082,11 @@ export class DeepLinkService implements Disposable {
 					return;
 				}
 				case DeepLinkServiceState.GoToTarget: {
-					// Need to re-fetch the remotes in case we opened in a new window
+					// Re-fetching the remotes in case we opened in a new window for specific actions
+					if (this._context.action === DeepLinkActionType.OpenAllPrChanges) {
+						action = DeepLinkServiceAction.OpenAllPrChanges;
+						break;
+					}
```

Comment:
Question here: If we skip straight to opening PR changes, wouldn't it fail to open the inspect view?

Logically, it should always open inspect regardless if a new window was opened or not, and then the action of opening PR changes can follow as a next step.

I would avoid modifying or using `this._context.action` directly because it uses that property outside of its intended purpose.

Reply:
>  If we skip straight to opening PR changes, wouldn't it fail to open the inspect view?

I don't understand your question. Current behavior is exactly how you described: open inspect, than if needed - open all changes (of course through ensuring repo)

> I would avoid modifying or using this._context.action

agree, but there is a new mechanic required. not sure if I'm able to suggest something reliable now, because I'm not well familiar with the deeplinking service. What do you think about a flag (such as `openedInNewWindow`) in the context, then we'll detect the flag and choose the right target. I'll push it now

Reply:
or there should be a middle state between ensureRepoMatch and goToTarget (e.g. repoMatchFinished) the goal of that is to redirect to goToTarget or to proceedNewWindowLink (the new one). If you like it, I'll implement. Here is a rude example:

```ts

case DeepLinkServiceState.GoToTarget: {
  /*leave as is*/
}
case DeepLinkServiceState.RepoMatchFinished: {
    if (this._context.isNewWindowOpened) {
       action = DeepLinkServiceAction.ProceedNewWindowLink
    } else {
       action = DeepLinkServiceAction.GoToTarget
    }
}
case DeepLinkServiceState.ProceedNewWindowLink: {
    if (this._context.action === DeepLinkActionType.SwitchToPullRequestWorktree) {
       action = DeepLinkServiceAction.OpenAllPrChanges
       break
    }
    action = DeepLinkServiceAction.DeepLinkResolved
}
```